File: font-end/src/service/Account/account.service.js
import httpClient from "../../api/http-comons";

export const numberOfCustomersLastMonth = () => {
  return httpClient.get(`/admin/account/numberOfCustomersLastMonth`);
};

export const numberOfCustomersThisMonth = () => {
  return httpClient.get(`/admin/account/numberOfCustomersThisMonth`);
};

export const listRoles = (role) => {
  return httpClient.get(`/admin/account/roles`);
};

export const login = (email, password) => {
  return httpClient.get(
    `/admin/account/login?email=${email}&password=${password}`
  );
};

export const detail = (id) => {
  return httpClient.get(`/admin/account/get-id?id=${id}`);
};

export const updateRole = (role, idUser) => {
  return httpClient.put(
    `/admin/account/updateRole?role=${role}&idUser=${idUser}`
  );
};

export const getAllAccount = () => {
  return httpClient.get(`/admin/account`);
};

File: back-end/src/main/java/com/example/backend/controller/order_management/service/AccountService.java
package com.example.backend.controller.order_management.service;

import com.example.backend.controller.order_management.model.account.AccountResponse;
import com.example.backend.entity.Account;

import java.util.List;

public interface AccountService {

    List<Account> findAll ();

    Account getOneByEmail(String email);

    AccountResponse getAccountUserByIdBill(String idBill );

    Integer numberOfCustomersLastMonth();

    Integer numberOfCustomersThisMonth();

//    Integer numberOfCustomers();

    Boolean updateAccount(String email, Integer id);

    Boolean updatePassword(String password, String passwordNew, String passwordRepeat, Integer id);

    Account getOneById(Integer id);

    Account updateRole(String role, Integer idUser);
}
